File: JS/Easy/lengthLastWord.js
var lengthLastWord = function(s) {
    var i = s.length - 1;
    var length = 0;

    // Skip trailing spaces
    while (i >= 0 && s[i] === ' ') {
        i--;
    }

    // Count the length of the last word
    while (i >= 0 && s[i] !== ' ') {
        length++;
        i--;
    }

    return length
}

// Examples
s1 = "Hello World"
s2 = "   fly me   to   the moon  "
s3 = "luffy is still joyboy"

console.log(lengthLastWord(s1))
console.log(lengthLastWord(s2))
console.log(lengthLastWord(s3))

File: Java/Easy/src/LengthLastWord.java
public class LengthLastWord {

    private int lengthLastWord(String s) {
        int i = s.length() - 1;
        int length = 0;

        // Skip trailing spaces
        while ( i >= 0 && s.charAt(i) == ' ') {
            i--;
        }

        // Skip trailing spaces
        while ( i >= 0 && s.charAt(i) != ' ') {
            length++;
            i--;
        }

        return length;
    }

    public static void main(String[] args) {
        LengthLastWord length = new LengthLastWord();

        String s1 = "Hell0 world";
        String s2 = "   fly me   to   the moon  ";
        String s3 = "luffy is still joyboy";
        System.out.println(length.lengthLastWord(s1));
        System.out.println(length.lengthLastWord(s2));
        System.out.println(length.lengthLastWord(s3));
    }
}
